### PR TITLE
chore: Add first cc unit tests.

### PR DESCRIPTION
**Pull request by mctavish**
This is the first step to bringing in a full suite of c++ unit tests for the Python debug agent.  For now, this contains only a test that does not need to link against the cpython api.

### CODE REVIEW

File: WORKSPACE [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  strip_prefix = "googletest-58d77fa8070e8cec2dc1ed015d66b454c8d78850",
+)
+
+# Used to build against Python.h
+http_archive(
+  name = "pybind11_bazel",
+  strip_prefix = "pybind11_bazel-faf56fb3df11287f26dbc66fdedf60a2fc2c6631",
+  urls = ["https://github.com/pybind/pybind11_bazel/archive/faf56fb3df11287f26dbc66fdedf60a2fc2c6631.zip"],
+)
+http_archive(
```
Comment (jasonborg) [inline WORKSPACE:41]: nit: add blank line

File: WORKSPACE [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+)
+
+http_archive(
+    name = "com_github_google_glog",
+    sha256 = "21bc744fb7f2fa701ee8db339ded7dce4f975d0d55837a97be7d46e8382dea5a",
+    strip_prefix = "glog-0.5.0",
+    urls = ["https://github.com/google/glog/archive/v0.5.0.zip"],
+)
+
+# Pinning to the 1.12.1; the last release that supports C++11
```
Comment (jasonborg) [inline WORKSPACE:28]: nit: missing 'release' after 1.12.1, or remove the leading 'the', and yes, I realize I'm the originator of this comment :)

File: cc-tests/bytecode_manipulator_test.cc [hunk 3]
```diff
@@ -0,0 +1,956 @@
+#include "src/googleclouddebugger/bytecode_manipulator.h"
```
Comment (jasonborg) [inline cc-tests/bytecode_manipulator_test.cc:1]: Missing Copyright

**Review by jasonborg — APPROVED** [review]

File: WORKSPACE (hunk 2 above)
Comment (mctavish) [reply WORKSPACE]: Done.

File: WORKSPACE (hunk 1 above)
Comment (mctavish) [reply WORKSPACE:42]: Done

File: cc-tests/bytecode_manipulator_test.cc (hunk 3 above)
Comment (mctavish) [reply cc-tests/bytecode_manipulator_test.cc]: Added.

**Comment by mctavish** [comment]
I moved the tests from /cc-tests to /tests/cpp and added a sentence to the README to point people to these tests.

File: README.md [hunk 4]
```diff
@@ -315,6 +315,9 @@ The following instructions are intended to help with modifying the codebase.
 Run the `build_and_test.sh` script from the root of the repository to build and
 run the unit tests using the locally installed version of Python.
 
+Run `bazel test tests/cpp:all` from the root of the repository to run unit
```
Comment (jasonborg) [inline README.md:318]: Do you also want to work this into the `build_and_test.sh` script?

**Review by jasonborg — APPROVED** [review]

File: README.md (hunk 4 above)
Comment (mctavish) [reply README.md:318]: I thought of that, but the need to install bazel to run the tests would change the requirements to run `build_and_test.sh`.  I'm going to leave them separate for now but may look into integrating the cpp tests into the `build-dist.sh` script.

**Pull request merged**